src/Microsoft.ServiceFabric.Services/Client/ServicePartitionResolver.cs

```diff
@@ -146,10 +151,7 @@ public ServicePartitionResolver(
         /// <param name="defaultServiceResolver">The new default value</param>
         public static void SetDefault(ServicePartitionResolver defaultServiceResolver)
         {
-            lock (StaticLock)
-            {
-                defaultResolver = defaultServiceResolver;
-            }
+            DefaultResolver = defaultServiceResolver;

```

Comment:
I don't think this is likely, but technically, I think locking was necessary because object variables are not guaranteed to be atomic: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/variables#96-atomicity-of-variable-references. I would use a lighter form of locking though: https://learn.microsoft.com/en-us/dotnet/api/system.threading.interlocked.exchange?view=net-8.0#system-threading-interlocked-exchange-1(-0@-0)

Reply:
As discussed offline, as this is a reference type, per the specification variable assignment is atomic, so this part should not be a problem.